resources/zib/zib-FunctionalOrMentalStatus.xml

```diff
@@ -0,0 +1,168 @@
+        <coding>
+          <system value="http://snomed.info/sct" />
+          <code value="384821006" />
+        </coding>
+      </patternCodeableConcept>
+    </element>
+    <element id="Observation.code">
+      <path value="Observation.code" />
+      <short value="StatusName" />
+      <definition value="Coded description of the functional or mental status or limitation. &#xD;&#xA;Code systems used include: &#xD;&#xA;-  SNOMED CT&#xD;&#xA;-  ICF (International Classification of Functioning, Disability and Health) &#xD;&#xA;-  NOC (Nursing Outcomes Classification)&#xD;&#xA;-  OMAHA" />
+      <alias value="StatusNaam" />
+      <binding>
+        <strength value="required" />
+        <valueSet value="http://decor.nictiz.nl/fhir/ValueSet/2.16.840.1.113883.2.4.3.11.60.121.11.4--20200901000000" />
+      </binding>
+      <mapping>
+        <identity value="zib-functionalormentalstatus-v3.2-2020EN" />
+        <map value="NL-CM:4.26.2" />
+        <comment value="StatusName" />
+      </mapping>
+    </element>
+    <element id="Observation.effective[x]">
+      <path value="Observation.effective[x]" />
+      <slicing>
+        <discriminator>
+          <type value="type" />
+          <path value="$this" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+    </element>
+    <element id="Observation.effective[x]:effectiveDateTime">
+      <path value="Observation.effective[x]" />
+      <sliceName value="effectiveDateTime" />
+      <short value="StatusDate" />
+      <definition value="Date on which the status was determined. A ‘vague’ date, such as only the year, is permitted." />
+      <alias value="StatusDatum" />
+      <type>
+        <code value="dateTime" />
+      </type>
+      <mapping>
+        <identity value="zib-functionalormentalstatus-v3.2-2020EN" />
+        <map value="NL-CM:4.26.6" />
+        <comment value="StatusDate" />
+      </mapping>
+    </element>
+    <element id="Observation.value[x]">
+      <path value="Observation.value[x]" />
+      <slicing>
+        <discriminator>
+          <type value="type" />
+          <path value="$this" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+    </element>
+    <element id="Observation.value[x]:valueCodeableConcept">
+      <path value="Observation.value[x]" />
+      <sliceName value="valueCodeableConcept" />
+      <short value="StatusValue" />
+      <definition value="The extent to which the patient is limited by the functional or mental condition. &#xD;&#xA;If both the status name and the status value are recorded in code, then both values must come from the same code system.  &#xD;&#xA;If the patient uses a medical device, the evaluation of the extent of the limitation will take place without this aid.&#xD;&#xA;The answer codes must be selected from the subselection of the code system corresponding with the request in accordance with the rules applicable to that system." />

```

Comment:
Should we include this as a constraint or is that too restricting?
Something like: if .code.coding.system and .valueCodeableConcept.coding.system are not empty, then .code.coding.system = .valueCodeableConcept.coding.system.

Reply:
A good suggestion! I would say it is not too restricting but it will require a complex invariant. I am not sure if that would outweigh the benefits. 

It will be complex because `.coding` is repeating and often multiple codings and systems are given in random order... you can only test for sure that both lists the `.code.coding.system` and the `.valueCodeableConcept.coding.system` need to have at least one system that is present in both lists. 

This was my first thoughts:
> (code.coding.system.exists() and valueCodeableConcept.coding.system.exists())  implies code.coding.where(system.subsetOf(valueCodeableConcept.coding.system)).exists()

But yeah again, I think we should skip this one. Unless you can shake something from you sleeve :p

Reply:
We could use the following expression:
`(code.coding.system.exists() and valueCodeableConcept.coding.system.exists()) implies (code.coding.system.distinct().subsetOf(valueCodeableConcept.coding.system.distinct()) and code.coding.system.distinct().supersetOf(valueCodeableConcept.coding.system.distinct()))`
By checking on both `subsetOf()` and `supersetOf()` we're actually checking equality of both sets. I've added `distinct()` because one might add multiple codes of the same code system. Do you think this constraint adds enough value to warrant such an expression? ;)

Reply:
Let's say we have for `.code` a coding in SNOMED and an additional coding in LOINC and for `.valueCodeableConcept` a coding in SNOMED. Then `.coding` is not a subset of `.valueCodeableConcept`. I think this is allowed though.